R-package/src/cmake/modules/FindLibR.cmake

```diff
@@ -174,17 +174,31 @@ execute_process(
   OUTPUT_VARIABLE LIBR_LIB_DIR
 )
 
+set(LIBR_HOME ${LIBR_HOME} CACHE PATH "R home directory")
+set(LIBR_EXECUTABLE ${LIBR_EXECUTABLE} CACHE PATH "R executable")
+set(LIBR_INCLUDE_DIRS ${LIBR_INCLUDE_DIRS} CACHE PATH "R include directory")
+set(LIBR_LIB_DIR ${LIBR_LIB_DIR} CACHE PATH "R shared libraries directory")
```

Comment:
In another PR, maybe we should change the name and use `LIBR_LIB_DIRS` instead of `LIBR_LIB_DIR` (cmake Find result variables should be plural for include_dirs / lib_dirs, usually they are placeholders for include_dir / lib_dir if they are singular directories).

Reply:
In this case it is a single directory. This isn't "all directories passed to the linker in `-L` flags", it is "what is the one directory where `R.dll` / `R.so`  is likely to live".

Anyway, I'm removing `LIBR_LIB_DIR` completely in #2963 . It's only being used in MSVC installs, and only indirectly as a way to find `LIBR_CORE_LIBRARY`.